server/src/main/java/org/opensearch/indices/IntKeyLookupStore.java

```diff
@@ -0,0 +1,164 @@
+
+    /**
+     * Attempts to safely remove a value from the internal structure, maintaining the property that contains(value)
+     * will never return a false negative. If removing would lead to a false negative, the value won't be removed.
+     * Classes may not implement safe removal.
+     * @param value The value to attempt to remove.
+     * @return true if the value was removed, false if it wasn't.
+     */
+    boolean remove(int value) throws Exception;
+
+    /**
+     * Check if the implementing class supports safe removals. If it doesn't, remove() will always return false.
+     * @return true if the class supports safe removals, false if it doesn't.
+     */
+    boolean supportsRemoval();
+
+    /**
+     * Remove the transformed version of this value from the store. Calling this function may cause
+     * contains() to return false negatives for future values.
+     * @param value The value to forcibly remove.
+     */
+    void forceRemove(int value) throws Exception;
+
+    /**
+     * Check if the object currently guarantees having no false negatives when running contains().
+     * @return false if there will not be false negatives, true if there could be false negatives.
+     */
+    boolean canHaveFalseNegatives();
+
+    /**
+     * Returns the number of distinct values stored in the internal data structure.
+     * Does not count values which weren't successfully added due to collisions.
+     * @return The number of values
+     */
+    int getSize();
+
+    /**
+     * Returns the number of times add() has been run, including unsuccessful attempts.
+     * @return The number of adding attempts.
+     */
+    int getNumAddAttempts();
+
+    /**
+     * Returns the number of times add() has returned false due to a collision.
+     * @return The number of collisions.
+     */
+    int getNumCollisions();
+
+    /**
+     * Returns the current internal data structure.
+     * @return A string representing the currently used internal data structure.
+     */
+    String getCurrentStructure() throws Exception;
+
+    /**
+     * If the structure uses a roaring bitmap with a modulo at some point, returns that modulo.
+     * If it uses a RBM without a modulo or doesn't use an RBM, returns 0.
+     * @return The modulo.
+     */
+    int getModulo();
+
+    /**
+     * Returns true if the transformation involves taking -abs(), simplifying int[] access and sorting
+     * @return Whether transformed values are always negative.
+     */
+    boolean isUsingNegativeOnly();
+
+    /**
+     * Checks if two values would collide after being transformed by this store's transformation.
+     * @param value1 The first value to compare.
+     * @param value2 The second value to compare.
+     * @return true if the transformations are equal, false otherwise.
+     */
+    boolean isCollision(int value1, int value2);
+
+    /**
+     * Returns an estimate of the store's memory usage.
+     * @return The memory usage, in MB
```

Comment:
Use bytes instead of MB. And long instead of double?
Also explicitly name function like getMemorySizeInBytes() to be clear.